source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,23 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
+  auto p = uriPattern.c_str();
+  while (*p && *uri && *p == *uri) {
+    p++;
+    uri++;
+  }
+  if (!(*p) && !(*uri)) { // strings are identical
+    return true;
+  }
+
+  if (*p == '*' && *(p - 1) == '/' && !(*(p + 1))) { // prefix matches
```

Comment:
I don't think you can always safely dereference `*(p - 1)`. Please add test for this while fixing.